docs/src/main/docs/guides/91_mp-tutorial.adoc

```diff
@@ -0,0 +1,1267 @@
+
+    // some lines omitted
+
+    @Path("/{name}")
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public JsonObject getMessage(@PathParam("name") String name) { <1>
+        return createResponse(name);
+    }
+    
+    @Path("/greeting")
+    @PUT
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response updateGreeting(JsonObject jsonObject) { <2>
+
+        if (!jsonObject.containsKey("greeting")) {
+            JsonObject entity = JSON.createObjectBuilder()
+                    .add("error", "No greeting provided")
+                    .build();
+            return Response.status(Response.Status.BAD_REQUEST).entity(entity).build();
+        }
+
+        String newGreeting = jsonObject.getString("greeting");
+
+        greetingProvider.setMessage(newGreeting);
+        return Response.status(Response.Status.NO_CONTENT).build();
+    }
+----
+
+<1> The first of these two methods implements a new HTTP GET service 
+ that returns JSON and it has a path parameter.  The `Path` annotation
+ defines the next part of the path to be a parameter named `name`.
+ In the method arguments the `PathParam("name")` annotation on 
+ `String name` has the effect of passing the parameter from the 
+ URL into this method as `name`.
+<2> The second method implements a new HTTP PUT service which produces
+ and consumes JSON, note the `Consumes` and `PUT` annotations.
+ It also defines a path of "/greeting".  Notice that the method
+ argument is a `JsonObject`. Inside the method body there is code
+ to check for the expected JSON, extract the value and update the
+ message in the `GreetingProvider`.
+
+Rebuild and run the application.  Test the new services using curl
+ commands similar to those shown below:
+
+[source,bash]
+.Testing the new services
+----
+curl -X GET http://localhost:8080/greet
+{"message":"Hello World!"}
+
+curl -X GET http://localhost:8080/greet/Joe
+{"message":"Hello Joe!"}
+
+curl -X PUT -H "Content-Type: application/json" -d '{"greeting" : "Hola"}' http://localhost:8080/greet/greeting
+
+curl -X GET http://localhost:8080/greet/Jose
+{"message":"Hola Jose!"}
+----
+
+Helidon MP provides many other features which can be added to the application.
+
+== Logging
+
+The application logging can be customized.  The default logging provider
+ is `java.util.logging`, however it is possible to use other providers.
+ In this tutorial the default provider is used.
+
+Create a `logging.properties` file in `src/main/resources` with
+ the following content:
+
+[source,java]
+.Example logging.properties file
+----
+# Send messages to the console
+handlers=java.util.logging.ConsoleHandler <1>
+
+# Global default logging level. Can be overriden by specific handlers and loggers
+.level=INFO <2>
+
+# Helidon Web Server has a custom log formatter that extends SimpleFormatter.
+# It replaces "!thread!" with the current thread name
+java.util.logging.ConsoleHandler.level=INFO
```

Comment:
It is almost always worth pointing out that if the `ConsoleHandler.level` is not set to, say, `FINER`, then no matter what you set a `Logger`'s level to the `ConsoleHandler` won't let it "through".  I personally set this to `FINER` then forget about it; then all I have to remember to do is to adjust the individual logger levels.